### PR TITLE
Blog app - creating a data model

### PR DESCRIPTION
**Pull request by AlaaShaheen2**
In this project I did:

- [ ]     Created a new app. Give it a meaningful name.
- [ ]     Initialize my project with Git.
- [ ]     Made sure that my project has Postgres database set up.
- [ ]     Build my project schema.
- [ ]         Created and run the necessary migration files.
- [ ]         Table and column names match the ERD diagram.
- [ ]         Foreign keys are included.
- [ ]         All columns that are foreign keys have a corresponding index.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,234 @@
...
+  <ul>
+    <li><a href="https://www.postgresql.org/">PostgreSQL</a></li>
+  </ul>
+</details>
+
+<!-- Features -->
+
+### Key Features <a name="key-features"></a>
+
+> Describe between 1-3 key features of the application.
```
Comment (FrederickMih) [inline README.md]: - This information was included in the README template in order to enlighten you on how to customize headers. This information is enough to let someone knows you used a template for your README file which is unprofessional 👎 . Kindly remove all these description blockquotes in your entire file.

File: db/migrate/20230128093911_create_user.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreateUser < ActiveRecord::Migration[7.0]
+  def change
+    create_table :users do |t|
+      t.string :name
+      t.string :photo
+      t.string :bio
```
Comment (FrederickMih) [inline db/migrate/20230128093911_create_user.rb:6]: - [ ] data type for `bio` should be text and not string.  The difference between the two is how many characters you can put in these fields. `string` has less character limit as compared to `text. `

File: db/migrate/20230128094327_create_comment.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class CreateComment < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.string :text
```
Comment (FrederickMih) [inline db/migrate/20230128094327_create_comment.rb:4]: - [ ] Kindly refer to the comment above ⬆️

File: db/migrate/20230128094549_create_post.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class CreatePost < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.string :text
```
Comment (FrederickMih) [inline db/migrate/20230128094549_create_post.rb:5]: - [ ] Kindly check the above hints

File: db/schema.rb [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+ActiveRecord::Schema[7.0].define(version: 2023_01_28_095151) do
+  # These are extensions that must be enabled in order to support this database
+  enable_extension "plpgsql"
+
+  create_table "comments", force: :cascade do |t|
+    t.string "text"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "post_id", null: false
```
Comment (FrederickMih) [inline db/schema.rb:21]: - [ ] It seems you haven't yet referenced the author commenting on this post

File: db/schema.rb [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  end
+
+  create_table "likes", force: :cascade do |t|
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "post_id", null: false
+    t.index ["post_id"], name: "index_likes_on_post_id"
+  end
+
+  create_table "posts", force: :cascade do |t|
```
Comment (FrederickMih) [inline db/schema.rb:36]: - [ ] We need the column of `author_id` in this table that will reference the user creating the post

File: db/schema.rb [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "post_id", null: false
+    t.index ["post_id"], name: "index_comments_on_post_id"
+  end
+
+  create_table "likes", force: :cascade do |t|
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "post_id", null: false
```
Comment (FrederickMih) [inline db/schema.rb:30]: - [ ] Let's kindly refer to the comments above. We need the user to like this post(author_id in this table)

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi 🙋 @AlaaShaheen2 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights

- PR is professional 👍 
- All Linter checks passed 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/schema.rb [hunk 8]
```diff
@@ -10,12 +10,12 @@
...
+ActiveRecord::Schema[7.0].define(version: 2023_01_28_112856) do
   # These are extensions that must be enabled in order to support this database
   enable_extension "plpgsql"
 
   create_table "comments", force: :cascade do |t|
-    t.string "text"
+    t.text "text"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
     t.bigint "post_id", null: false
```
Comment (MahmoudBakr23) [inline db/schema.rb:21]: - [ ] Please add the reference for the author in both comments and posts tables as requested by the previous reviewer. 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hi @AlaaShaheen2 😃, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BertrandConxy — APPROVED** [review]
Hi @AlaaShaheen2, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💪 
- No linter errors
- project requirements are met
- project works fine
- good PR description
- correct flow is used

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**